src/language-js/printer-estree.js

```diff
@@ -6300,17 +6345,17 @@ function printAssignment(
 
   return group(concat([printedLeft, operator, printed]));
 }
-
-function adjustClause(node, clause, forceSpace) {
+// [prettierx] breakBeforeStatement option
+function adjustClause(node, clause, hardBreak, forceSpace) {
```

Comment:
nit: I am actually not so convinced to insert a new `hardBreak` argument before the existing `forceSpace` argument. This could cause a future merge from upstream Prettier to suddenly go wrong if they start using `adjustClause` with `forceSpace` value someday. Yes it is probably a bit cleaner this way but I am not super comfortable with it.

The ultimate solution may be to consider using an options object for the forceSpace and hardBreak options. Maybe good for a future enhancement.

Reply:
@brodybits sorry for not polling this that often. Let me know what suits your need the best:

1. `forceSpace` before `hardBreak`